cosign/verifier.go

```diff
@@ -0,0 +1,544 @@
+		return nil, fmt.Errorf("error unmarshaling json: %w", err)
+	}
+	if jsonData == nil {
+		return nil, fmt.Errorf("unmarshaled json data is nil")
+	}
+
+	// 2. Get the log index, log ID, integrated time, signed entry timestamp and body
+	payload, ok := jsonData["Payload"].(map[string]any)
+	if !ok || payload == nil {
+		return nil, fmt.Errorf("error getting Payload")
+	}
+
+	logIndex, ok := payload["logIndex"].(float64)
+	if !ok {
+		return nil, fmt.Errorf("error getting logIndex")
+	}
+	li, ok := payload["logID"].(string)
+	if !ok {
+		return nil, fmt.Errorf("error getting logID")
+	}
+	logID, err := hex.DecodeString(li)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding logID: %w", err)
+	}
+	integratedTime, ok := payload["integratedTime"].(float64)
+	if !ok {
+		return nil, fmt.Errorf("error getting integratedTime")
+	}
+	set, ok := jsonData["SignedEntryTimestamp"].(string)
+	if !ok {
+		return nil, fmt.Errorf("error getting SignedEntryTimestamp")
+	}
+	signedEntryTimestamp, err := base64.StdEncoding.DecodeString(set)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding signedEntryTimestamp: %w", err)
+	}
+	// 3. Unmarshal the body and extract the rekor KindVersion details
+	body, ok := payload["body"].(string)
+	if !ok {
+		return nil, fmt.Errorf("error getting body")
+	}
+	bodyBytes, err := base64.StdEncoding.DecodeString(body)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding body: %w", err)
+	}
+	err = json.Unmarshal(bodyBytes, &jsonData)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshaling json: %w", err)
+	}
+	if jsonData == nil {
+		return nil, fmt.Errorf("unmarshaled body json data is nil")
+	}
+	apiVersionRaw, ok := jsonData["apiVersion"]
+	if !ok || apiVersionRaw == nil {
+		return nil, fmt.Errorf("error getting apiVersion")
+	}
+	apiVersion, ok := apiVersionRaw.(string)
+	if !ok {
+		return nil, fmt.Errorf("apiVersion is not a string")
+	}
+	kindRaw, ok := jsonData["kind"]
+	if !ok || kindRaw == nil {
+		return nil, fmt.Errorf("error getting kind")
+	}
+	kind, ok := kindRaw.(string)
+	if !ok {
+		return nil, fmt.Errorf("kind is not a string")
+	}
+	// 4. Construct the transparency log entry list
+	return []*protorekor.TransparencyLogEntry{
+		{
+			LogIndex: int64(logIndex),
+			LogId: &protocommon.LogId{
+				KeyId: logID,
+			},
+			KindVersion: &protorekor.KindVersion{
+				Kind:    kind,
+				Version: apiVersion,
+			},
+			IntegratedTime: int64(integratedTime),
+			InclusionPromise: &protorekor.InclusionPromise{
+				SignedEntryTimestamp: signedEntryTimestamp,
+			},
+			InclusionProof:    nil,
+			CanonicalizedBody: bodyBytes,
+		},
+	}, nil
+}
+
+// getBundleMsgSignature returns the bundle message signature from the simple signing layer
+func getBundleMsgSignature(simpleSigningLayer ocispec.Descriptor) (*protobundle.Bundle_MessageSignature, error) {
+	// 1. Get the message digest algorithm
+	var msgHashAlg protocommon.HashAlgorithm
+	switch simpleSigningLayer.Digest.Algorithm() {
```

Comment:
consider
```suggestion
	switch alg := simpleSigningLayer.Digest.Algorithm(); alg {
```

so that it won't be called again at line 450